rust_src/src/keymap.rs

```diff
@@ -50,6 +85,86 @@ pub fn keymapp(object: LispObject) -> bool {
     map.is_not_nil()
 }
 
+/// Return the parent map of KEYMAP, or nil if it has none.
+/// We assume that KEYMAP is a valid keymap.
+#[no_mangle]
+pub extern "C" fn keymap_parent(keymap: Lisp_Object, autoload: bool) -> Lisp_Object {
+    let map = unsafe { LispObject::from_raw(get_keymap(keymap, true, autoload)) };
```

Comment:
My only concern here is both `find` and `last` will walk the list. So this will be a little less efficient than the previous hand coded loop. Do we care? I do not know.

Reply:
Hmm. I think this would be a concern for really large keymaps. And keymaps can get really large sometimes. Actually my old implementation has a `find` and `tail` too. I'll try looking for a different approach, if I can come up with one.

Reply:
So I changed the implementation to this:

```rust
#[no_mangle]
pub extern "C" fn keymap_parent(keymap: Lisp_Object, autoload: bool) -> Lisp_Object {
    let map = unsafe { LispObject::from_raw(get_keymap(keymap, true, autoload)) };
    let mut current = LispObject::constant_nil();
    for elt in map.iter_tails_safe() {
        current = elt.cdr();
        if keymapp(current) {
            return current.to_raw();
        }
    }
    unsafe { get_keymap(current.to_raw(), false, autoload) }
}
```

The idea is that it only needs to walk the map once with `map.iter_tails_safe` not unlike the previous implementation where it would go through `find` and `last`. Tell me what you think. Thanks!

Reply:
Also, I looked at my first implementation and it is similar to this new one that I made. So I guess, if efficiency is concerned, this would be the way to go. ;)

Reply:
Yeah, as much as I like the other version this is the better choice.